### PR TITLE
SK-145/ README for Upsert support

### PR DESCRIPTION
**Pull request by Shaik-Luqmaan**
Signed-off-by: Shaik-Luqmaan <luqmaan.shaik@skyflow.com>

### CODE REVIEW

File: samples/vaultapi/upsert.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func main() {
+
+	defer func() {
+		if err := recover(); err != nil {
+			fmt.Println("error : ", err)
+		}
+	}()
+	logger.SetLogLevel(logger.INFO) //set loglevel to INFO
```
Comment (CourtneyDocs) [inline samples/vaultapi/upsert.go:39]: Update comment to sentence structure: Set loglevel to INFO.

File: README.md [hunk 2]
```diff
@@ -389,11 +450,11 @@ Sample response:
 }
 ```
 
-### Invoke-connection
+### InvokeConnection
 
 Using  InvokeConnection, end-user applications can integrate checkout/card issuance flow with their apps/systems. To invoke connection, use the invokeConnection(config ConnectionConfig) method of the Skyflow client. The config object must have `connectionURL`,`methodName` and remaining are optional.
```
Comment (CourtneyDocs) [inline README.md:455]: Change slashes to "and" instead. Checkout and card issuance...apps and systems.

File: README.md [hunk 3]
```diff
@@ -389,11 +450,11 @@ Sample response:
...
 
-Using the InvokeConnection method, you can integrate their server-side application with third party APIs and services without directly handling sensitive data. Prior to invoking the `InvokeConnection` method, you must have created a connection and have a connectionURL already generated. Once you have the connectionURL, you can invoke a connection by using the **invokeConnection(config ConnectionConfig)** method. The config parameter must include a `connectionURL` and `methodName`. The other fields are optional.
+Using the InvokeConnection method, you can integrate their server-side application with third party APIs and services without directly handling sensitive data. Prior to invoking the `InvokeConnection` method, you must have created a connection and have a connectionURL already generated. Once you have the connectionURL, you can invoke a connection by using the **InvokeConnection(config ConnectionConfig)** method. The config parameter must include a `connectionURL` and `methodName`. The other fields are optional.
```
Comment (CourtneyDocs) [inline README.md:457]: Who is "their" on line 457?
I would update the following sentences to read: 

The InvokeConnection method lets you bypass handling sensitive data by integrating (replace their) server-side application with third-party APIs.

Prior to invoking the `InvokeConnection` method, you must create a connection and generate a connectionURL.

File: README.md [hunk 4]
```diff
@@ -289,9 +350,9 @@ Sample response:
 }
 ```
 
-#### Get By Id
+#### GetById
  
-In order to retrieve data from your vault using SkyflowIDs, use the **getById(records map[string]interface{})** method. The `records` parameter takes a map that should contain an array of SkyflowIDs to be fetched, as shown below:
+In order to retrieve data from your vault using SkyflowIDs, use the **GetById(records map[string]interface{})** method. The `records` parameter takes a map that should contain an array of SkyflowIDs to be fetched, as shown below:
```
Comment (CourtneyDocs) [inline README.md:355]: I would change, The `records` parameter takes a map that should contain an array of SkyflowIDs to be fetched, as shown below: 

to

The `records` parameter takes a map that has an array of SkyflowIDs to return, as shown below:

File: README.md [hunk 5]
```diff
@@ -201,13 +206,69 @@ Sample response :
...
+        "cardNumber": "f37186-e7e2-466f-91e5-48e2bcbc1",
+        "fullname": "1989cb56-63a-4482-adf-1f74cd1a5"
+      }
+    }
+  ]
+}
+```
+
+#### Detokenize
+In order to retrieve data from your vault using tokens that you have previously generated for that data, you can use the **Detokenize(records map[string]interface{})** method. The first parameter must have a `records` key that takes an array of tokens to be fetched from the vault, as shown below.
```
Comment (CourtneyDocs) [inline README.md:271]: I'd update 

"In order to retrieve data from your vault using tokens that you have previously generated for that data, you can use the **Detokenize(records map[string]interface{})** method," 

to

To retrieve tokens from your vault,  you can use...

The records parameter takes a map that has an array of SkyflowIDs to return, as shown below:

File: README.md [hunk 6]
```diff
@@ -112,16 +112,19 @@ skyflowClient := Skyflow.Init(configuration)
 
 All Vault APIs must be invoked using a skyflowClient instance.
 
-#### Insert
+### Insert data into the vault
+
+To insert data into your vault, use the **Insert(records map[string]interface{}, options common.InsertOptions)** method of the Skyflow client. The **insertInput** parameter requires a `records` key and takes an array of records to be inserted into the vault as a value. The `options` parameter is a InsertOptions object that provides further options, including Upsert operations, for your insert call, as shown below.
```
Comment (CourtneyDocs) [inline README.md:117]: Change 

> The **insertInput** parameter requires a `records` key and takes an array of records to be inserted into the vault as a value.

to active voice:

The **insertInput** parameter requires a `records` key and takes an array of records to insert as a value into the vault.

File: README.md (hunk 2 above)
Comment (CourtneyDocs) [inline README.md:455]: Change:

> Using InvokeConnection, end-user applications can integrate checkout/card issuance flow with their apps/systems.

End-user apps can use InvokeConnection to integrate checkout and card issuance flows with their apps and systems.

> To invoke connection

To invoke a connection

> and remaining are optional.

and the remaining are optional.

**Review by CourtneyDocs — CHANGES_REQUESTED** [review]
LGTM. The main nitpicks are changing the voice and making sentences more concise.

File: samples/vaultapi/upsert.go (hunk 1 above)
Comment (Shaik-Luqmaan) [reply samples/vaultapi/upsert.go:39]: Done

File: README.md (hunk 3 above)
Comment (Shaik-Luqmaan) [reply README.md:457]: "their" refers to any third-party services

File: README.md (hunk 6 above)
Comment (Shaik-Luqmaan) [reply README.md:117]: Done

File: README.md (hunk 2 above)
Comment (Shaik-Luqmaan) [reply README.md:455]: Done

**Review by yaswanth-pula-skyflow — APPROVED** [review]

**Review by yennamsanthosh-skyflow — APPROVED** [review]

**Pull request merged**